### PR TITLE
Imposing "correct" isort as requirement for a green pipeline

### PR DESCRIPTION
**Pull request by andped10**
The CI pipeline will now fail if python imports are done "wrongly" according to isort.

The requirements for the `isort` sorting are defined in `pyproject.toml`
- all python files have been sorted accordingly

It is also possible to get the requirements applied on file save in VSCode by using the `settings.json` file in `vscode_template`.

To help the user in case of a failing pipeline the following now step echo's a potential resolution to the error.  This is not an optimal solution, but it is considered good enough for now.

Bumped:
- actions/setup-python@v5 
- actions/checkout@v4

### CODE REVIEW

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -15,8 +15,17 @@ name: CI using pip
...
+  CI_Code_Consistency:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: isort/isort-action@v1
+      - name: Suggestion to fix errors
+        if: ${{ failure() }}
+        run: |
+            echo "::notice::In project root run 'python.exe -m isort .' and commit changes to fix errors."
+            exit 1
```
Comment (rozyczko) [inline .github/workflows/python-ci.yml:27]: Is failure the right thing to do? Maybe stylistic issues should only be reported as warnings, like it is done with `codecov` reports?

File: pyproject.toml [hunk 2]
```diff
@@ -59,15 +63,21 @@ documentation = "https://docs.easyreflectometry.org"
...
 source = ['EasyReflectometry']
 
 [tool.github.info]
 organization = 'easyScience'
 repo = 'EasyReflectometryLib'
 
-[tool.hatch.build.targets.wheel]
-packages = ["EasyReflectometry"]
+[tool.isort]
+line_length = 88
```
Comment (rozyczko) [inline pyproject.toml:77]: This is probably too conservative. With hi res monitors, we tend to write long lines, especially for message formatting.
We specify 120 as the standard max length.
Any good reason to limit the width to 88 (apart from this being the standard in `black`?)

**Review by rozyczko — CHANGES_REQUESTED** [review]
The code looks and behaves well but let's have a discussion about standardising items like 
- line width
- import layout
- hard failure on static code analysis

File: .github/workflows/python-ci.yml (hunk 1 above)
Comment (andped10) [reply .github/workflows/python-ci.yml:27]: We can do it differently.  The reason why I decided to go for an exit 1 of the echo step is an attempt to make it more obvious that the user might want to consult the produced statement that contains a proposed fix to the issue.

The pipeline itself has already failed in the preceding step.

We should probably discuss.

File: pyproject.toml (hunk 2 above)
Comment (andped10) [reply pyproject.toml:77]: It was just a value that frequently showed up.

Monkey see, monkey do  ;)

**Pull request merged**